sdk/javascript/generator/StructTypeFormatter.py

```diff
@@ -129,6 +129,29 @@ def get_ctor_descriptor(self):
 
 		return MethodDescriptor(body=body, arguments=arguments)
 
+	def get_comparer_descriptor(self):
+		if not self.struct.comparer:
+			return None
+
+		body = ''
+		if any('ripemd_keccak_256' == transform for (_, transform) in self.struct.comparer):
+			body += 'const { ripemdKeccak256 } = require(\'../utils/transforms\'); // eslint-disable-line global-require\n\n'
+
+		body += 'return [\n'
+		for (property_name, transform) in self.struct.comparer:
+			body += '\t'
+			if not transform:
+				body += f'this.{lang_field_name(property_name)}'
+			else:
+				body += f'{lang_field_name(transform).replace("_", "")}(this.{lang_field_name(property_name)}.bytes)'
```

Comment:
given `.bytes` I assume this handles only array-like types now, right?

Reply:
correct

Reply:
should that go in a `.md` (not sure, asking)

Reply:
i think we would add support for non-array, if needed, so would not add to md